src/main/java/com/exasol/adapter/commontests/scalarfunction/ScalarFunctionsTestBase.java

```diff
@@ -143,550 +129,449 @@ private void runOnExasol(final ExasolExecutable exasolExecutable) {
+                new DataTypeWithExampleValue(DataType.createDouble(), 0.5),
+                new DataTypeWithExampleValue(DataType.createDecimal(18, 0), 2),
+                new DataTypeWithExampleValue(DataType.createBool(), true),
+                new DataTypeWithExampleValue(DataType.createVarChar(2, DataType.ExaCharset.UTF8), "a"),
+                new DataTypeWithExampleValue(DataType.createDate(), new Date(1000)),
+                new DataTypeWithExampleValue(DataType.createTimestamp(false), new Timestamp(1001)));
+        private VirtualSchemaTestSetup virtualSchemaTestSetup;
 
         @BeforeAll
-        void beforeAll() throws SQLException {
-            this.testTable = createDateVirtualSchemaTable();
-            this.queryBuilder = new ScalarFunctionQueryBuilder(this.testTable.getFullyQualifiedName());
-            this.columnName = getColumnsOfTable(this.testTable.getFullyQualifiedName()).get(0);
+        void beforeAll() {
+            this.virtualSchemaTestSetup = createTableWithExampleValues();
+            runOnExasol(statement -> {
+                statement.executeUpdate("CREATE SCHEMA " + LOCAL_COPY_SCHEMA);
+                final String testTable = this.virtualSchemaTestSetup.getFullyQualifiedName() + ".\"" + MY_TABLE + "\"";
+                statement
+                        .executeUpdate("CREATE TABLE " + LOCAL_COPY_FULL_TABLE_NAME + " as SELECT * FROM " + testTable);
+                this.columnsWithType = getColumnsOfTable(LOCAL_COPY_FULL_TABLE_NAME);
+                this.parameterCache = new ScalarFunctionsParameterCache();
+                this.parameterFinder = new ScalarFunctionParameterFinder(this.columnsWithType,
+                        new ScalarFunctionQueryBuilder(LOCAL_COPY_FULL_TABLE_NAME), this.parameterCache);
+                this.virtualSchemaRunVerifier = new VirtualSchemaRunVerifier(new ScalarFunctionQueryBuilder(testTable));
+            });
         }
 
-        @AfterAll
-        void afterAll() {
-            this.testTable.drop();
+        private VirtualSchemaTestSetup createTableWithExampleValues() {
+            final List<Column> columns = this.DATA_TYPES_WITH_EXAMPLES.stream()
+                    .map(dataTypeWithExampleValue -> new Column(dataTypeWithExampleValue.getExasolDataType().toString(),
+                            getCachedTestSetup()
+                                    .getDataTypeThatThisAdapterMapsTo(dataTypeWithExampleValue.getExasolDataType())))
+                    .collect(Collectors.toList());
+            final List<Object> rowWithExampleValues = this.DATA_TYPES_WITH_EXAMPLES.stream()
+                    .map(DataTypeWithExampleValue::getExampleValue).collect(Collectors.toList());
+            final VirtualSchemaTestSetupRequest request = new VirtualSchemaTestSetupRequest(
+                    TableRequest.builder(MY_TABLE).columns(columns).row(rowWithExampleValues).build());
+            return getCachedTestSetup().getVirtualSchemaTestSetupProvider()
```

Comment:
I find it a bit confusing that here we use the cached test setup but in line 145 we use the actual one.
I thought the cached one was read-only and the other one for writing but I guess not.

When should we use one and when the other? Maybe we can make this distinction explicit somehow?